hipe4ml/TrainingAndTesting/plot_utils.py

```diff
@@ -0,0 +1,310 @@
+    fig = plt.figure(figsize=(20, 10))
+    # plt.title(t,y=1.08,fontsize=16)
+    plt.suptitle(tit, fontsize=18, ha='center')
+    grid = ImageGrid(fig, 111, nrows_ncols=(1, 2), axes_pad=0.15, share_all=True,
+                     cbar_location='right', cbar_mode='single', cbar_size='7%', cbar_pad=0.15)
+
+    opts = {'cmap': plt.get_cmap(
+        'coolwarm'), 'vmin': -1, 'vmax': +1, 'snap': True}
+
+    ax1 = grid[0]
+    ax2 = grid[1]
+    heatmap1 = ax1.pcolor(corrmat_sig, **opts)
+    heatmap2 = ax2.pcolor(corrmat_bkg, **opts)
+    ax1.set_title('Signal', fontsize=14, fontweight='bold')
+    ax2.set_title('Background', fontsize=14, fontweight='bold')
+
+    lab = corrmat_sig.columns.values
+    for axs in (ax1,):
+        # shift location of ticks to center of the bins
+        axs.set_xticks(np.arange(len(lab)), minor=False)
+        axs.set_yticks(np.arange(len(lab)), minor=False)
+        axs.set_xticklabels(lab, minor=False, ha='left',
+                            rotation=90, fontsize=10)
+        axs.set_yticklabels(lab, minor=False, va='bottom', fontsize=10)
+        axs.tick_params(axis='both', which='both', direction="in")
+
+        for tick in axs.xaxis.get_minor_ticks():
+            tick.tick1line.set_markersize(0)
+            tick.tick2line.set_markersize(0)
+            tick.label1.set_horizontalalignment('center')
+
+    for axs in (ax2,):
+        # shift location of ticks to center of the bins
+        axs.set_xticks(np.arange(len(lab)), minor=False)
+        axs.set_yticks(np.arange(len(lab)), minor=False)
+        axs.set_xticklabels(lab, minor=False, ha='left',
+                            rotation=90, fontsize=10)
+        axs.tick_params(axis='both', which='both', direction="in")
+        for tick in axs.xaxis.get_minor_ticks():
+            tick.tick1line.set_markersize(0)
+            tick.tick2line.set_markersize(0)
+            tick.label1.set_horizontalalignment('center')
+
+    ax1.cax.colorbar(heatmap1)
+    ax2.cax.colorbar(heatmap2)
+    ax1.cax.toggle_label(True)
+    return fig
+
+
+def plot_bdt_eff(threshold, eff_sig):
+    """Plot the BDT efficiency calculated with the function bdt_efficiency_array() in training_utils.py
+
+    Input
+    -----------------------------------
+    threshold: score threscold array
+    eff_sig: bdt efficiency array
+
+    Output
+    -----------------------------------
+    res: matplotlib object of the bdt efficiency as a function of the threshold score
+
+    """
+    res = plt.figure()
+    plt.plot(threshold, eff_sig, 'r.', label='Signal efficiency')
+    plt.legend()
+    plt.xlabel('BDT Score')
+    plt.ylabel('Efficiency')
+    plt.title('Efficiency vs Score')
+    plt.grid()
+    return res
+
+
+def plot_roc(y_truth, model_decision):
+    """Calculate and plot the roc curve
+
+    Input
+    -------------------------------------
+    y_truth: test set label
+    model decision: predicted score
```

Comment:
What type they should be? Also it looks to me that here the model prediction should be a probability.